### PR TITLE
[ NEW FEATURE ] Sign Up page

### LINKED ISSUE #44: Signup page

### PR DESCRIPTION
**Pull request by banahaker**
# [ NEW FEATURE ] Sign Up page

Create new Signup Page

Fixes #44 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/BlockSignUpPage.vue [hunk 1]
```diff
@@ -1,3 +1,344 @@
...
+  password: string;
+  email: string;
+}
+const signUpFormData = reactive<loginData>({
+  username: '',
+  password: '',
+  email: '',
+});
+
+const oauthButtons = [
```
[the remark this answers is not in the document]
Comment (banahaker) [reply src/components/BlockSignUpPage.vue]: Oops i forgot to delete it

File: src/components/BlockSignUpPage.vue [hunk 2]
```diff
@@ -1,3 +1,338 @@
+<script lang="ts" setup>
+import { reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
+import ToolLangSelector from './ToolLangSelector.vue';
+const { t } = useI18n();
```
[the remark this answers is not in the document]
Comment (banahaker) [reply src/components/BlockSignUpPage.vue]: oh fxck i forgot twice

File: src/components/BlockSignUpPage.vue [hunk 3]
```diff
@@ -1,3 +1,337 @@
+<script lang="ts" setup>
+import { reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
```
[the remark this answers is not in the document]
Comment (banahaker) [reply src/components/BlockSignUpPage.vue]: ...

File: src/components/BlockSignUpPage.vue [hunk 4]
```diff
@@ -1,3 +1,336 @@
+<script lang="ts" setup>
+import { reactive } from 'vue';
+import ToolLangSelector from './ToolLangSelector.vue';
+
+interface loginData {
```
Comment (SiongSng) [inline src/components/BlockSignUpPage.vue]: it is `signUpData` not `loginData`

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: src/components/BlockSignUpPage.vue (hunk 4 above)
Comment (banahaker) [reply src/components/BlockSignUpPage.vue]: oh sorry

**Comment by SiongSng** [comment]
please fix

**Comment by banahaker** [comment]
Fixed

**Review by SiongSng — APPROVED** [review]
LGTM!

**Pull request merged**